plugins/catalog-backend/src/ingestion/ConfigGenerator.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '@backstage/catalog-model';
+import { Logger } from 'winston';
+import { ConfigGenerator } from './types';
+
+export class ConfigGeneratorClient implements ConfigGenerator {
```

Comment:
The word "config" is overloaded, usually referring to the actual config of the app. May want to use combinations of "location" and "analyze" in my opinion - `LocationAnalyzer`, with a method `analyzeLocation` that takes an `AnalyzeLocationRequest` etc

Reply:
I agree that the word "Config" is too generic, but generating is *what* this feature will do, analyzing path is only the *how*. Other names I was considering was combinations of "Component|Entity", "Info|Config" and then "Generator|Creator". Maybe you would like to have this name more like that? (either way, you have better knowledge on how it interacts with other names in the codebase, so just choose it, I will adapt to it)

Reply:
Changed to LocationAnalyzer